packages/sync-server/src/sync/updateLookupTable.ts

```diff
@@ -23,7 +23,9 @@ const updateLookupTableForModel = async (
 
   let fromId = '';
   let totalCount = 0;
-  const result: SyncLookupQueryDetails = (await model.buildSyncLookupQueryDetails()) || {};
+  const result: SyncLookupQueryDetails = Boolean(model.buildSyncLookupQueryDetails)
+    ? (await model.buildSyncLookupQueryDetails()) || {}
+    : {};
```

Comment:
Perhaps this is more straightforward?

```suggestion
  const result: SyncLookupQueryDetails = model.buildSyncLookupQueryDetails?.() ?? {};
```


Alternatively, a more explicit version of your approach, since the cast to a boolean happens even without calling `Boolean()`:

```suggestion
  const result: SyncLookupQueryDetails = typeof model.buildSyncLookupQueryDetails === 'function'
    ? (await model.buildSyncLookupQueryDetails()) || {}
    : {};
```

</details>

Reply:
Oh yep that's much better! Thanks for that!

This change is also meant to be done in the base branch, but I accidentally commit it in this branch. I'll update and move it back to the base branch so this PR only contains changes about syncDirection!